src/ir/attributes/double_attribute.h

```diff
@@ -0,0 +1,60 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#ifndef SRC_IR_ATTRIBUTES_DOUBLE_ATTRIBUTE_H_
+#define SRC_IR_ATTRIBUTES_DOUBLE_ATTRIBUTE_H_
+
+#include "absl/strings/str_format.h"
+#include "src/common/utils/intrusive_ptr.h"
+#include "src/ir/attributes/attribute.h"
+
+namespace raksha::ir {
+
+class DoubleAttribute;
+bool operator==(const DoubleAttribute& lhs, const DoubleAttribute& rhs);
+
+class DoubleAttribute : public AttributeBase {
+ public:
+  static constexpr Kind kAttributeKind = Kind::kDouble;
+
+  static intrusive_ptr<const DoubleAttribute> Create(double value) {
+    return intrusive_ptr<const DoubleAttribute>(new DoubleAttribute(value));
+  }
+
+  double value() const { return value_; }
+
+  bool IsEqual(const AttributeBase& other) const override {
+    if (this->kind() != other.kind()) return false;
+    return *this == static_cast<const DoubleAttribute&>(other);
+  }
+
+  std::string ToString() const override {
+    return absl::StrFormat("%gl", value_);
```

Comment:
I think you want `%lg`. [According to the docs for printf](https://en.cppreference.com/w/cpp/io/c/fprintf) the length modifier is the second-to-last character code in the print format, followed only by the format specifier itself. `g` is the format specifier, meaning "floating point in exponential notation" and `l` is the length specifier, meaning "long" or "double".

Reply:
Actually, I think you want `%a`. `%g` goes out to a fixed precision, whereas `%a` goes out to the minimum precision needed to exactly represent the value.

Reply:
Here `l` is a suffix meant to ensure that double values like `3.0` are not accidentally converted to `int` (as %g prints them as just `3`).

I'll test `%lgl` though.

Reply:
Hmm, looking at the doc I think `%a` is exclusively hexadecimal. So far the best results I've seen come from `%g` and `%lg` (I'll use lg for now)